src/Analyzers/Orchestration/DateTimeOrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis;
+using Microsoft.DurableTask.Analyzers.Helpers;
+
+namespace Microsoft.DurableTask.Analyzers.Orchestration;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class DateTimeOrchestrationAnalyzer : OrchestrationAnalyzer
+{
+    public const string DiagnosticId = "DURABLE0001";
+
+    static readonly DiagnosticDescriptor Rule = new(
+        DiagnosticId,
+        "DateTime calls must be deterministic inside an orchestration function",
+        "The method '{0}' uses '{1}' that may cause non-deterministic behavior when invoked from Orchestration Function '{2}'",
+        AnalyzersCategories.Orchestration,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => [Rule];
+
+    protected override void RegisterAdditionalCompilationStartAction(CompilationStartAnalysisContext context, OrchestrationAnalysisResult orchestrationAnalysisResult)
+    {
+        INamedTypeSymbol systemDateTimeSymbol = context.Compilation.GetSpecialType(SpecialType.System_DateTime);
+
+        ConcurrentBag<(ISymbol symbol, IPropertyReferenceOperation operation)> dateTimeUsage = [];
```

Comment:
Can we add a comment here describing how `dateTimeUsage` is used?

Reply:
done :)